### DISCUSSION TITLE
hiddenon for table columns

### DISCUSSION THREAD

**Discussion by ankorinek** [filamentphp/filament#6623] [Ideas]
I have a resource table that I would like to reuse in a relation manager class. However one field should be hidden when the table is used in that context. For forms there is the hiddenon property, which does exactly that. I haven't seen a similar function for table columns. Does this already exist and could someone please point me to the docs for that?

**Comment by pboivin** [comment]
Hi @ankorinek, not sure if there's a better hook but I think this should work : 
```php
    ->hidden(fn() => static::class !== MyResource::class)
```

**Comment by ankorinek** [comment]
Hi pboivin,

that does not work unfortunately, static::class always returns the resource class regardless if it's called within the relationmanager or not.

**Reply by pboivin** [reply]
Ah yes, makes sense... Would it work if you extracted the shared `table()` method into a trait?

**Comment by danharrin** [comment]
`->hidden(fn ($livewire) => $livewire::class === RelationManagerClass::class)`

**Reply by ankorinek** [reply]
This works, thanks.

**Reply by clementmas** [reply]
This seems to work as well:
```php
->hidden(fn ($livewire) => str_contains($livewire::class, 'RelationManagers'))
```
